packages/js/src/utils/id-token.ts

```diff
@@ -0,0 +1,39 @@
+import { UrlSafeBase64 } from '@silverhand/essentials';
+import * as s from 'superstruct';
+
+/**
+ * @link [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)
+ */
+const IdTokenClaimsSchema = s.type({
+  iss: s.string(),
+  sub: s.string(),
+  aud: s.string(),
+  exp: s.number(),
+  iat: s.number(),
+  at_hash: s.optional(s.string()),
+});
+
+export type IdTokenClaims = s.Infer<typeof IdTokenClaimsSchema>;
+
+export const decodeIdToken = (token: string): IdTokenClaims => {
+  const { 1: encodedPayload } = token.split('.');
+
+  if (!encodedPayload) {
+    throw new Error('invalid token');
+  }
+
+  const json = UrlSafeBase64.decode(encodedPayload);
+
+  try {
+    // Use Superstruct to validate the json type
+    const idToken = JSON.parse(json) as IdTokenClaims;
+    s.assert(idToken, IdTokenClaimsSchema);
+    return idToken;
```

Comment:
why do we still have `JSON.parse` here?

Reply:
rewrote like this, PTAL:
```ts
  const json = UrlSafeBase64.decode(encodedPayload);
  return s.create(json, IdTokenClaimsSchema);
}
```

Reply:
I think the `json` value from `UrlSafeBase64.decode` is a string. Not sure we could bypass the JSON.parse.

Reply:
Still use `JSON.parse` according to JOSE implementation.

Reply:
The variable named `json` can also present a string value containing json content. e.g. demo code from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

![image](https://user-images.githubusercontent.com/10594507/150109125-c88a07cc-5aab-4874-aae4-3610b9a5a4b0.png)